### PR TITLE
feat: [CDC-16046]: Manifest collection support build WF

### PR DESCRIPTION
**Pull request by puthrayaharness**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/29721)

### CODE REVIEW

**Review by poojaSinghal — APPROVED** [review]

**Review by poojaSinghal — DISMISSED** [review]
> Reviewed 15 of 185 files at r1, 170 of 170 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/29721)* status: :shipit: complete! ✓

**Review by poojaSinghal — APPROVED** [review]
> Reviewed 34 of 34 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/29721)* status: :shipit: complete! ✓

File: 400-rest/src/main/java/io/harness/serializer/kryo/ManagerKryoRegistrar.java [hunk 1]
```diff
@@ -1700,5 +1703,8 @@ public void register(Kryo kryo) {
     kryo.register(Type.class, 400131);
     kryo.register(OSType.class, 400132);
     kryo.register(HelmChartCollectionParams.HelmChartCollectionType.class, 400134);
+    kryo.register(ServiceHelmElement.class, 400135);
```
Comment (puthrayaharness) [inline 400-rest/src/main/java/io/harness/serializer/kryo/ManagerKryoRegistrar.java:1706]: changed kryo pojo but not sending it in tasks

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 14 of 185 files at r1, 3 of 34 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/29721)* status: :shipit: complete! ✓

**Pull request merged**